Use Plugin::load() when creating classless plugins.

markstory opened the pull request:
I completely missed that Plugin::load() does a bunch of work to generate the correct paths for plugins. By using Plugin::load() and fetching the created plugin we can leverage the same logic. There is an argument to be had around deprecating Plugin::load(). Given the troubles plugin classes have given us I'm not comfortable in doing that right now.

Refs #12271

markstory committed:
Use Plugin::load() when creating classless plugins.

I completely missed that Plugin::load() does a bunch of work to generate
the correct paths for plugins. By using Plugin::load() and fetching the
created plugin we can leverage the same logic. There is an argument to
be had around deprecating Plugin::load(). Given the troubles plugin
classes have given us I'm not comfortable in doing that right now.

Refs #122271

saeideng commented:
can we duplicate code ?
 `Plugin::load()` has problem with calling bootstrap directly

markstory commented:
Darn I forgot about that. Duplicating code will allow us to deprecate plugin::load() more easily in the future as well I guess.

markstory committed:
Move plugin configuration file loading to PluginCollection

Moving the loading of this file here allows us Plugin to get slimmer
making `Plugin::load()` easier to deprecate later. It also solves
a problem where the plugin configuration wouldn't be loaded if an
application used all new style plugins. While this might not matter,
I suspect there is an application somewhere that does.

I've also moved plugin path resolution into PluginCollection. This makes
sharing logic between the two plugin paths easier preventing code
duplication.

I've opted to make BaseApplication::makePlugin() protected as
I mistakenly made it public before.

markstory commented:
@saeideng I think I've resolved the issue you raised and also fixed a potential problem where `cakephp-plugins.php` would not get loaded.

markstory commented on src/Http/BaseApplication.php:
```diff
@@ -119,22 +123,22 @@ public function getPlugins()
      * @return \Cake\Core\PluginInterface
      */
-    public function makePlugin($name, array $config)
+    protected function makePlugin($name, array $config)
```
There is no reason this method needs to be public, and I consider it a mistake that I would like to fix now vs. later. If folks think it is a risky change I'm ok with reverting this.

ADmad replied:
Making is protected sounds good to me.
